test/unit/db/repos/peers.js

```diff
@@ -0,0 +1,200 @@
+			.seed(db)
+			.then(() => done(null))
+			.catch(done);
+	});
+
+	afterEach(done => {
+		sinonSandbox.restore();
+		seeder
+			.reset(db)
+			.then(() => done(null))
+			.catch(done);
+	});
+
+	it('should initialize db.delegates repo', () => {
+		return expect(db.peers).to.be.not.null;
+	});
+
+	describe('PeersRepository', () => {
+		describe('constructor()', () => {
+			it('should assign param and data members properly', () => {
+				expect(db.peers.db).to.be.eql(db);
+				return expect(db.peers.pgp).to.be.eql(db.$config.pgp);
+			});
+		});
+
+		describe('list()', () => {
+			it('should use the correct SQL with no params', function*() {
+				sinonSandbox.spy(db, 'any');
+				yield db.peers.list();
+
+				expect(db.any.firstCall.args[0]).to.eql(peersSQL.list);
+				return expect(db.any.firstCall.args[1]).to.be.eql(undefined);
+			});
+
+			it('should return list of peers', function*() {
+				const peers = [];
+
+				for (let i = 0; i < 5; i++) {
+					const peer = peersFixtures.DBPeer();
+					const peer2 = Object.assign({}, peer);
+					peer2.broadhash = Buffer.from(peer2.broadhash, 'hex');
+					yield db.query(
+						db.$config.pgp.helpers.insert(peer2, null, { table: 'peers' })
+					);
+					peers.push(peer);
+				}
+
+				const result = yield db.peers.list();
+
+				expect(result).to.be.not.empty;
+				expect(result).to.have.lengthOf(5);
+				expect(result[0]).to.have.all.keys(
+					'ip',
+					'wsPort',
+					'state',
+					'os',
+					'version',
+					'broadhash',
+					'height',
+					'clock'
+				);
+				return expect(result).to.be.eql(peers);
+			});
+		});
+
+		describe('clear()', () => {
+			it('should use the correct SQL with no params', function*() {
+				sinonSandbox.spy(db, 'none');
+				yield db.peers.clear();
+
+				expect(db.none.firstCall.args[0]).to.eql(peersSQL.clear);
+				return expect(db.none.firstCall.args[1]).to.be.eql(undefined);
+			});
+
+			it('should clear all peers and resolve with empty response', function*() {
+				for (let i = 0; i < 5; i++) {
+					const peer = peersFixtures.DBPeer();
+					peer.broadhash = Buffer.from(peer.broadhash, 'hex');
+					yield db.query(
+						db.$config.pgp.helpers.insert(peer, null, { table: 'peers' })
+					);
+				}
+
+				const before = yield db.one('SELECT count(*) from peers;');
+				const result = yield db.peers.clear();
+				const after = yield db.one('SELECT count(*) from peers;');
+
+				expect(before.count).to.be.eql('5');
+				expect(result).to.be.null;
+				return expect(after.count).to.be.eql('0');
+			});
+		});
+
+		describe('insert()', () => {
+			it('should use the correct query method with correct params', function*() {
+				// Few lines are commented due to issue in pg-promise
+				// https://github.com/vitaly-t/pg-promise/issues/473
+
+				sinonSandbox.spy(db, 'none');
+				// sinonSandbox.spy(db.peers.pgp.helpers, 'insert');
+				const peer = peersFixtures.DBPeer();
+				yield db.peers.insert(peer);
+
+				return expect(db.none).to.be.calledOnce;
```

Comment:
Insufficient test. It should attempt to locate the inserted peer instead, and see that was in fact added.